realm/realm-library/src/main/java/io/realm/RealmChangeListener.java

```diff
@@ -29,19 +29,19 @@
  * for that background thread.
  * <p>
  * All {@link io.realm.RealmObject} and {@link io.realm.RealmResults} will automatically contain their new values when
- * the {@link #onChange()} method is called. Normally this means that it isn't necessary to query again for those
+ * the {@link #onChange(Object)} method is called. Normally this means that it isn't necessary to query again for those
  * objects, but just invalidate any UI elements that are using them. If there is a chance that a object has been been
  * deleted, it can be verified by using {@link RealmObject#isValid()}.
  *
  * @see Realm#addChangeListener(RealmChangeListener)
  * @see Realm#removeAllChangeListeners()
  * @see Realm#removeChangeListener(RealmChangeListener)
  */
-public interface RealmChangeListener {
+public interface RealmChangeListener<T> {
 
     /**
      * Called when a transaction is committed.
      */
-    void onChange();
+     void onChange(T object);
```

Comment:
we should rule out anythings involving change/update as the Object (i.e RealmResult/Object) may not be changed. _until we have fine grained notifications_

How about `element`?